Add tooltip to map 875

inuwa opened the pull request:
## Done

Added tooltip text showing percentages in the "Where are Ubuntu users" section of the /desktop/statistics page.
## QA 

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8001/](http://0.0.0.0:8001/)
- Run through the following [QA steps](https://github.com/canonical-webteam/practices/blob/master/workflow/qa-steps.md)
- Hover over the map in the "Where are Ubuntu Users" section and see if the tooltip appears over the country.


## Issue / Card

Fixes https://github.com/ubuntudesign/web-squad/issues/875
## Screenshots

![image](https://user-images.githubusercontent.com/43535482/48353359-23c6ba80-e687-11e8-8af5-65f3c0715809.png)

webteam-app commented:
Starting demo at: https://www-ubuntu-com-canonical-websites-pr-4376.run.demo.haus/

anasereijo commented:
@inuwa this is starting to look great! just 3 things:
- some countries don't show a tooltip (for example Brazil and Australia);
- on the tooltip can we match the top margin with the bottom one so the text is vertically centred?
- tooltips on the left and right sides of the page appear cropped (see below) maybe we can make them `class="p-tooltip p-tooltip--top-center"` or `class="p-tooltip p-tooltip--right"
`
<img width="528" alt="screenshot 2018-11-12 at 14 37 57" src="https://user-images.githubusercontent.com/36884067/48354100-c7fd3100-e688-11e8-97df-367028cbfe05.png">

anthonydillon commented:
QA issues:
- When resizing the screen the map is redrawn which causes them to scroll.
- I think a top tooltip would be better instead of a left aligned one.
- There are a few counties what do not get a hover. I.e. Brazil, etc.
- The horizontal scroll does not work on small screens anymore.

anthonydillon commented:
I get scrollbars on Firefox:
![screenshot_2018-11-13 user statistics ubuntu](https://user-images.githubusercontent.com/1413534/48416473-3654f880-e748-11e8-9fbb-2f64d14b9b93.png)

inuwa commented:
@anthonydillon sorry not sure but the scroll bars do not show up on my desktop.
![image](https://user-images.githubusercontent.com/43535482/48418024-23dcbe00-e74c-11e8-9602-b7fd296e74f5.png)
I'll try a workaround though.

webteam-app commented:
Starting demo at: https://www-ubuntu-com-canonical-websites-pr-4376.run.demo.haus/

anthonydillon commented:
There are lot's of issues when scaling the screen.

anasereijo commented:
Hi @inuwa,

I think you forgot to address one of my previous comments:
- [ ] on the tooltip can we match the top margin with the bottom one so the text is vertically centred?

And I have a few more comments:

- [ ] Can we match the left and right paddings on the tooltips? It doesn't seem very consistent now. For example, on Russia seems ok but then on Brazil there is more padding on the right side. 
<img width="487" alt="screenshot 2018-11-13 at 15 57 12" src="https://user-images.githubusercontent.com/36884067/48425547-f77d6d80-e75c-11e8-8855-adc4f07d1455.png">
<img width="984" alt="screenshot 2018-11-13 at 15 55 15" src="https://user-images.githubusercontent.com/36884067/48425548-f8ae9a80-e75c-11e8-951d-9236c9e51548.png">


- [ ] As @anthonydillon said there are lot's of issues with scaling the screen, for example, when I make my viewport smaller the map disappears. I need to refresh the browser to see the map again (see image below)
<img width="1664" alt="screenshot 2018-11-13 at 15 51 13" src="https://user-images.githubusercontent.com/36884067/48425257-5bebfd00-e75c-11e8-9b36-119a1a594695.png">


- [ ] In order to see the map on my phone I have to refresh the browser



- [ ] On my phone, if I click on a country the map disapears

inuwa commented:
@anthonydillon I have added a workaround for the scroll issue in Ubuntu's version of firefox. Hopefully it works.
@anasereijo I have added the padding to the tooltip text. Cheers

anthonydillon commented:
QA issues:
1. I still get scrollbars on desktop. There are two elements with `overflow: auto;`
![screenshot_2018-11-14 user statistics ubuntu](https://user-images.githubusercontent.com/1413534/48473434-5a6c1480-e7f0-11e8-952c-12a510670901.png)

2. Canada and Greenlands tooltip is not positioned correctly

3. On smaller screens there are two scrollbars:
![screenshot_2018-11-14 user statistics ubuntu 1](https://user-images.githubusercontent.com/1413534/48473537-8daea380-e7f0-11e8-85aa-acacf7b83851.png)

Greggless commented:
In interest of keeping numbers secret, and for making it _slightly_ easier for the user to read, could we round the values to the nearest 0.1%.  

If a value is less that 0.1% can it be displayed as <0.1%

The colouring of the map should also reflect this.

Thanks

# 🐟

inuwa commented:
@anthonydillon Good morning. I have removed the scroll bars in firefox for large screens. Cheers
@Greggless the labels have now been customised for less than 1% and border lines added now.
@anasereijo the tooltips now have equal padding left and right now.

anthonydillon commented:
I see see the double scrollbars on small screens

anthonydillon commented:
Awesome! Thank you for sticking with this one. LGTM :+1:. Can you rebase the the commits on this branch into a single commit please.

inuwa commented:
> Awesome! Thank you for sticking with this one. LGTM 👍. Can you rebase the the commits on this branch into a single commit please.

@anthonydillon you are welcome. I have compressed the commits also. Cheers

anasereijo commented:
Hi @inuwa this is looking good!

I'm giving +1 but as discussed feel free to find out what is causing that weird tooltip on top of the scale.

![User statistics   Ubuntu.png](https://images.zenhubusercontent.com/5a9531ab4b5806bc2bc88d30/8edbc1a3-f37d-4188-bf28-4da9326885b8)

inuwa commented:
@anasereijo thank you. I have corrected the issue now
